### DISCUSSION TITLE
Problem with the state

### DISCUSSION THREAD

**Discussion by agudelodiego** [vercel/next.js#44035] [Help]
### Summary

Hello, my name is Diego, i'm a front end developer. I have a litle problem: I'm building my porfolio using next js and server side rendering. I have a state wich is an array of objects, each object represent comments who came from a backend, and for any reason when i try to update the state, this is not updated. I like know why this happens, because when i do same thing in react it work very well. Maybe you could contact, or send me a email explain this, and how can i solve this problem.

### Additional information

`import Styles from '../styles/Chat.module.css'
import Comment from './Comment'
import NewComment from './NewComment'
import { useEffect, useState } from 'react'

//! Interfaces
import {comment} from '../interfaces'

interface props { 
  comments: comment[],
  _id: string
}

const Chat = ({comments,_id}:props) => {
  const [feed,setFeed] = useState(comments);

  const pushComment = async(newcomment:comment)=>{
    let swap = feed
    swap.push(newcomment)
    setFeed(swap)
    console.log('Ejecucion de push')
    console.log(feed)
  }

  useEffect(()=>{
    console.log('Acrtualizando el estado')
    console.log(feed)
  },[feed])

  return (
    <section className={Styles.chatContainer}>
      <h2 className={Styles.cgreen}>Hazme un comentario</h2>
      <div className={Styles.chat}>
        {feed.map((comment:comment)=>{return <Comment comment={comment} key={comment._id}/>})}
      </div>
      <NewComment _id={_id} pushComment={pushComment}/>
    </section>
  )
}
export default `Chat``

**Comment by ciruz** [comment]
Hey do you have an example repo ?

**Comment by agudelodiego** [comment]
Hey tom, whatsapp men ? how are you?. Tom don't worry about it, i solved
the problem.
But actually, when i was doing the deployment, i had another issue:

Description:
So, you know, in next js we have some ways to use images, one of those are
import our image in our code like a component, something like this:
[image: image.png]
[image: image.png]
When i do that, it doesn't work at all well, because when i used this
approach and then open up my local server and go to section where the image
should be, I find that
[image: image.png]
Then want to show you if i try to change the import to uppercase, it work
[image: image.png]
[image: image.png]
But the strange thing come when i open up my file browser and see this
imagine and note that
[image: image.png]
The image name doesn't have uppercase letter but however it's work.



Tom actually in my repo solved this problem just using other approach in my
main branch, but if you want to check the problem i gonna create a
branch called bug where i gonna put this code in /components/Main.tsx
So, the repo url is: https://github.com/agudelodiego/Porfolio
If you have problems running the server just tell me.
For run the repo you need to put in the root folder a .env file with this
variable:
SANITY_STUDIO_TOKEN=skRo5jak7IZ70k1OW4NhTCPzxkWOF3OtBsn7vFlto5O0H9saEYLh9iaHP90WsSfTymmKcH26oMY32lcNnltAC5UlgDqvVCCSIimAIvKW6C8cEtalXATNDc3ihv1O0EI5IjKcrSetqlAVY0x9wP5ST2X84kIdkXP1u7TiLfduRUBgmZnJf33l


I'm waiting for your response. Bye
